src/ContainerRenameModal.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import {
+    Button,
+    Form, FormGroup,
+    Modal, TextInput
+} from '@patternfly/react-core';
+import cockpit from 'cockpit';
+
+import * as client from './client.js';
+import { ErrorNotification } from './Notification.jsx';
+
+const _ = cockpit.gettext;
+
+const ContainerRenameModal = (props) => {
+    const [name, setName] = useState(props.container.Names[0]);
+    const [nameError, setNameError] = useState(null);
+    const [dialogError, setDialogError] = useState(null);
+    const [dialogErrorDetail, setDialogErrorDetail] = useState(null);
+
+    const handleInputChange = (targetName, value) => {
+        if (targetName === "name") {
+            setName(value);
+            setNameError(null);
+        }
+    };
+
+    const handleRename = () => {
+        if (!name) {
+            setNameError(_("Container name is required"));
+            return;
+        }
+
+        const renameData = {
+            name: name
+        };
+        setNameError(null);
+        setDialogError(null);
+        client.renameContainer(props.container.isSystem, props.container.Id, renameData)
```

Comment:
Please do this inline, with `renameData` → `{ name }`; more succinct, and avoids the indirection with the single-use constant.

`{ x }` is JS shorthand for `{ x: x }`, BTW, as it's a very common use case.